universal-application-tool-0.0.1/app/views/FileUploadStrategy.java

```diff
@@ -0,0 +1,192 @@
+    return applicantQuestionRendererFactory.getRenderer(question).render(params);
+  }
+
+  private Tag renderFileKeyField(
+      ApplicantQuestion question, ApplicantQuestionRendererParams params) {
+    return FileUploadQuestionRenderer.renderFileKeyField(question, params, false);
+  }
+
+  private Tag renderEmptyFileKeyField(
+      ApplicantQuestion question, ApplicantQuestionRendererParams params) {
+    return FileUploadQuestionRenderer.renderFileKeyField(question, params, true);
+  }
+
+  Tag renderFileUploadBottomNavButtons(Params params) {
+    Optional<Tag> maybeContinueButton = maybeRenderContinueButton(params);
+    Optional<Tag> maybeSkipOrDeleteButton = maybeRenderSkipOrDeleteButton(params);
+    ContainerTag ret =
+        div()
+            .withClasses(ApplicantStyles.APPLICATION_NAV_BAR)
+            // An empty div to take up the space to the left of the buttons.
+            .with(div().withClasses(Styles.FLEX_GROW))
+            .with(renderReviewButton(params));
+    if (maybeSkipOrDeleteButton.isPresent()) {
+      ret.with(maybeSkipOrDeleteButton.get());
+    }
+    ret.with(renderUploadButton(params));
+    if (maybeContinueButton.isPresent()) {
+      ret.with(maybeContinueButton.get());
+    }
+    return ret;
+  }
+
+  private Tag renderReviewButton(Params params) {
+    String reviewUrl =
+        routes.ApplicantProgramReviewController.review(params.applicantId(), params.programId())
+            .url();
+    return a().attr(HREF, reviewUrl)
+        .withText(params.messages().at(MessageKey.BUTTON_REVIEW.getKeyName()))
+        .withId("review-application-button")
+        .withClasses(ApplicantStyles.BUTTON_REVIEW);
+  }
+
+  /**
+   * Renders a form submit button for continue form if an uploaded file is present.
+   *
+   * <p>See {@link renderDeleteAndContinueFileUploadForms}.
+   */
+  private Optional<Tag> maybeRenderContinueButton(Params params) {
+    if (!hasUploadedFile(params)) {
+      return Optional.empty();
+    }
+    Tag button =
+        submitButton(params.messages().at(MessageKey.BUTTON_KEEP_FILE.getKeyName()))
+            .attr(FORM, FILEUPLOAD_CONTINUE_FORM_ID)
+            .withClasses(ApplicantStyles.BUTTON_BLOCK_NEXT)
+            .withId("fileupload-continue-button");
+    return Optional.of(button);
+  }
+
+  /**
+   * Renders a form submit button for delete form if the file upload question is optional.
+   *
+   * <p>If an uploaded file is present, render the button text as delete. Otherwise, skip.
+   *
+   * <p>See {@link renderDeleteAndContinueFileUploadForms}.
+   */
+  private Optional<Tag> maybeRenderSkipOrDeleteButton(Params params) {
+    if (hasAtLeastOneRequiredQuestion(params)) {
+      // If the file question is required, skip or delete is not allowed.
+      return Optional.empty();
+    }
+    String buttonText = params.messages().at(MessageKey.BUTTON_SKIP_FILEUPLOAD.getKeyName());
+    String buttonId = "fileupload-skip-button";
```

Comment:
nit: Same here, this and line 161 should probably be constants instead of hard-coded strings